install/Knative-with-Minishift.md

```diff
@@ -0,0 +1,250 @@
+
+1. Set up the project **myproject** for use with Knative applications.
+
+    ```shell
+    oc project myproject
+    oc adm policy add-scc-to-user privileged -z myproject
+    oc label namespace myproject istio-injection=enabled
+    ```
+    The `oc adm policy` adds the **privileged** [Security Context Constraints(SCCs)](https://docs.okd.io/3.10/admin_guide/manage_scc.html) to the **default** Service Account. The SCCs are the precursor to the PSP (Pod Security Policy) mechanism in Kubernetes.
+
+    Its is also ensured that the project myproject is labelled for Istio automatic sidecar injection, with this `istio-injection=enabled` label to **myproject** each of the Knative applications that will be deployed in **myproject** will have Istio sidecars injected automatically. 
+
+  > **IMPORTANT:** Avoid using `default` project in OpenShift for deploying Knative applications. As OpenShift deploys few of its mission critical applications in `default` project, its safe not to touch to avoid any crash to the OpenShift.
+
+### Installing Istio
+
+Knative depends on Istio. The [istio-openshift-policies.sh](scripts/istio-openshift-policies.sh) does run the required commands to configure necessary [privileges](https://istio.io/docs/setup/kubernetes/platform-setup/openshift/) to the service accounts used by Istio.
+
+```shell
+bash <(curl -s https://raw.githubusercontent.com/knative/docs/master/install/scripts/istio-openshift-policies.sh)
+```
+
+1. Run the following to install Istio:
+
+    ```shell
+    curl -L https://storage.googleapis.com/knative-releases/serving/latest/istio.yaml \
+    | oc apply -f -
+    ```
+    > **NOTE:** If you get a lot of errors after running the above command like:  __unable to recognize "STDIN": no matches for kind "Gateway" in version "networking.istio.io/v1alpha3"__, just run the command above again, it's idempotent and hence objects will be created only once.
+
+2. Ensure the istio-sidecar-injector pods runs as provileged:
+    ```shell
+    oc get cm istio-sidecar-injector -n istio-system -o yaml  \
+     | sed -e 's/securityContext:/securityContext:\\n      privileged: true/' \
+     | oc replace -f -
+    ```
+3. Monitor the Istio components until all of the components show a `STATUS` of `Running` or `Completed`:
+
+    ```shell
+    while oc get pods -n istio-system | grep -v -E "(Running|Completed|STATUS)"; do sleep 5; done
+    ```
+> **NOTE:** It will take a few minutes for all the components to be up and running. Use CTRL+C to exit watch mode
+
+## Install Knative Serving
+
+The following section details on deploying [Knative Serving](https://github.com/knative/serving) to OpenShift.
+
+The [knative-openshift-policies.sh](scripts/knative-openshift-policies.sh) runs the required commands to configure necessary privileges to the service accounts used by Knative.
+
+```shell
+bash <(curl -s https://raw.githubusercontent.com/knative/docs/master/install/scripts/knative-openshift-policies.sh)
```

Comment:
This (the content of the shell script) gives me warnings:

```
Warning: ServiceAccount 'build-controller' not found
cluster role "cluster-admin" added: "build-controller"
Warning: ServiceAccount 'controller' not found
cluster role "cluster-admin" added: "controller"
```

Even IF ok... we should mention that